php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/StaticConstantAccess.java

```diff
@@ -54,10 +62,24 @@ public Identifier getConstantName() {
         while (expression instanceof ExpressionArrayAccess) {
             expression = ((ExpressionArrayAccess) expression).getExpression();
         }
+        // can use dynamic constant names since PHP 8.3
+        // e.g. Foo::{$bar};
+        assert expression != null;
+        if (isDynamicName) {
+            // XXX create an identifier to avoid NPE, can improve this?
+            if (expression instanceof ReflectionVariable) {
+                expression = ((ReflectionVariable) expression).getName();
+            }
+            expression = new Identifier(constant.getStartOffset(), constant.getEndOffset(), "{" + expression.toString() + "}"); // NOI18N
+        }
         assert expression instanceof Identifier;
```

Comment:
Perhaps a bit better?

`assert expression instanceof Identifier : "Expected Identifier but got: " + expression.getClass().getName();`

Reply:
Yes, I'll change to it :) Thanks!